src/clj/puppetlabs/services/ca/certificate_authority_service.clj

```diff
@@ -31,14 +34,18 @@
           host-crl-file (.getCanonicalPath (fs/file
                                             (get-in-config [:puppetserver :hostcrl])))
           infra-nodes-file (.getCanonicalPath (fs/file (str (fs/parent ca-crl-file) "/infra_inventory.txt")))
-          watcher (create-watcher {:recursive false})]
+          watcher (create-watcher {:recursive false})
+          report-activity-or-nil (if-let [activity-reporting-service (maybe-get-service this :ActivityReportingService)]
+                                   (partial activity-proto/report-activity! activity-reporting-service)
+                                   nil)
```

Comment:
Just a thought, but you could use `(constantly nil)` to provide a no-op function in place of the `nil` here, an then the "report-activity" could always be a function.  That might reduce the cognitive burden of tracking it a bit?  Would also simplify `report-cert-event` slightly.  

https://clojuredocs.org/clojure.core/constantly